### PR TITLE
fix broken link

### PR DESCRIPTION
**Pull request by nojnhuh**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:

This PR fixes the failing link checker on an old k8s v1.25 link that is no longer active.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [X] cherry-pick candidate <-- I imagine this same check will begin failing on release branches too

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/book/src/developers/development.md [hunk 1]
```diff
@@ -538,7 +538,7 @@ With the following environment variables defined, you can build a CAPZ cluster f
 | Environment Variable      | Value                                                                                                                                                                                                    |
 | ------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | `E2E_ARGS`                | `-kubetest.use-ci-artifacts`                                                                                                                                                                             |
-| `KUBERNETES_VERSION`      | `latest` - extract Kubernetes version from https://dl.k8s.io/ci/latest.txt (main's HEAD)<br>`latest-1.25` - extract Kubernetes version from https://dl.k8s.io/ci/latest-1.25.txt (release branch's HEAD) |
+| `KUBERNETES_VERSION`      | `latest` - extract Kubernetes version from https://dl.k8s.io/ci/latest.txt (main's HEAD)<br>`latest-1.<MINOR>` - extract Kubernetes version from dl.k8s.io/ci/latest-1.<MINOR>.txt (release branch's HEAD) |
```
Comment (nojnhuh) [inline docs/book/src/developers/development.md:541]: Removing `https://` seems to be enough to fool the link checker.

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Pull request merged**